plugins/rapid7_vulndb/komand_rapid7_vulndb/util/extract.py

```diff
@@ -25,7 +36,10 @@ def execute_query(cls, query: Dict) -> Dict:
         :return: Data as dictionary
         """
         response = requests.get(cls.search_url, params=query, allow_redirects=False)
-        response.raise_for_status()
+        if response.status_code != requests.codes.ok:
+            time.sleep(5)
+            response = requests.get(cls.search_url, params=query, allow_redirects=False)
+        response_error_handler(response.status_code, response.text)
```

Comment:
I think you can just move `time.sleep` value to a separate variable. It'd be much easier in the future to change if needed :smile: Well, I think it's possible to move that part of code to a private method (because you are using same code at lines 124:127 as well) but it's up to you.
```suggestion
        if response.status_code != requests.codes.ok:
            time.sleep(DELAY_TIME)
            response = requests.get(cls.search_url, params=query, allow_redirects=False)
        response_error_handler(response.status_code, response.text)
```
As I can see retry function just work once after bad status code response before. Wouldn't be better to use such as exponential back-off function to retry request in a specific number of attempts?

Reply:
Removed time.sleep from method and replaced with retry method with exponential backoff.